src/ethers/wrapper.ts

```diff
@@ -253,12 +253,27 @@ class WalletWrapper {
   async getGasPrice (): Promise<BigNumber> {
     let gasPrice: BigNumber
     if (this.estimateGasPrice) {
-      const providerGasPrice: any = BigNumber.from(
-        await this.provider.getGasPrice()
-      )
-      gasPrice = BigNumber.from(
-        Math.ceil(providerGasPrice * this.gasPriceFactor)
-      )
+      try {
+        const factor: number = Math.ceil(this.gasPriceFactor * 100)
+        const providerGasPrice: any = BigNumber.from(
+          await this.provider.getGasPrice()
+        )
+        gasPrice = BigNumber.from(
+          Math.ceil(providerGasPrice * this.gasPriceFactor)
+        )
+        gasPrice = gasPrice.mul(factor).div(100)
```

Comment:
I'm a bit confused here. Let's say that:
- `this.gasPriceFactor` is `1.25`
- `this.provider.getGasPrice()` gives `500`

Then:
- `factor` will be `125`
- `providerGasPrice` will be `500`
- `gasPrice @ 261` will take value `625`
- `gasPrice @ 264` will take value `781.25`

Maybe I'm missing something, but it looks to me like we're wrongly using the factor twice. One of the two multiplications may be redundant, I think :thinking:

Reply:
Factor multiplication method will now be unified on both getGasPrice and getGasLimit functions